js/okx.js

```diff
@@ -4700,16 +4703,23 @@ module.exports = class okx extends Exchange {
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
-        const marginMode = this.safeStringLower (params, 'mgnMode');
-        params = this.omit (params, [ 'mgnMode' ]);
+        let marginMode = undefined;
+        [ marginMode, params ] = this.handleMarginModeAndParams ('setLeverage', params);
+        if (marginMode === undefined) {
+            marginMode = this.safeString (params, 'mgnMode', 'cross'); // cross as default marginMode
+        }
         if ((marginMode !== 'cross') && (marginMode !== 'isolated')) {
-            throw new BadRequest (this.id + ' setLeverage() params["mgnMode"] must be either cross or isolated');
+            throw new BadRequest (this.id + ' setLeverage() requires a marginMode parameter that must be either cross or isolated');
         }
         const request = {
             'lever': leverage,
             'mgnMode': marginMode,
             'instId': market['id'],
         };
+        const posSide = this.safeString (params, 'posSide');
```

Comment:
@Dan-krm, actually we don't need to do this:

```Javascript
        if (posSide !== undefined) {
            request['posSide'] = posSide;
        }
```
because it will be done automatically upon extending the request with params.

I meant only this:
```Javascript
const posSide = this.safeString (params, 'posSide');
if (marginMode === 'isolated' && posSide === undefined) {
    throw (...)
}
```
but don't worry I'll fix it this time